### ISSUE TITLE
Modifying components to be non-updatable or have zero learning rate [+ more general configs]

### ISSUE THREAD

**Issue by danpovey** [kaldi-asr/kaldi#1040]
A user on the list was asking about the way to set an individual component's learning rate, e.g. to zero, after training for a bit, and I can see no existing mechanism to do this.

In order to change the learning rate of a specific node of an already trained model, the existing pattern would dictate that it should be done via an option to nnet3-am-copy or nnet3-copy (probably a function that would be added to nnet-utils.h that could be called by both).  That code does not seem to exist yet.  We need to write it.

But actually I wonder if it might be better to do this in a more general and extensible way.  My concept is to extend the config format the nnet3-init uses, and also to make it available as an option to nnet3-copy and nnet3-am-copy (--apply-config=xxx).

We already need to add the command
`delete-node name=[node-name]`
which would delete a node; and we could perhaps modify the ReadConfig() function so that it will automatically delete 'orphan' components in cases where the delete-node command had been used, so there would be no need for a 'delete-component' command [it might be a hassle to have to delete the component separately]  It already warns about orphan components, but it doesn't delete them.

We could also add commands like:

`set-learning-rate name=[component-name-pattern] (underlying-learning-rate=xxx|learning-rate-factor=xxx)`
and maybe, if needed,
`scale-learning-rate name=[component-name-pattern] (underlying-learning-rate-scale=xxx|learning-rate-factor-scale=xxx)`

and:
`convert-to-fixed-affine name=[component-name-pattern]`
which would convert a NaturalGradientAffineComponent (or similar) to a FixedAffineComponent... this is more efficient than setting the learning rate to zero as it avoids the backprop completely, if possible.

If we want to support the component-name patterns, it's tricky because it's only C++11 and we don't require C++11 yet.  What I actually think might be better is to only support `*`, like filename globs, so you could say layer*-affine or something like that. [E.g. add a function `bool NameMatchesPattern(const std::string &name, const std::string &pattern);` in nnet-parse.h.. a recursive implementation that works char by char and calls itself more than once would be very easy to implement].  This mechanism would certainly be easy to understand; and the lack of power could be worked around just by having people explicitly list the nodes they want to change, so there is no real loss of generality.

Comments are welcome on this issue.

**Comment by danpovey** [comment]
Update: I'd like to add another command
rename-node old-name=xxx new-name=xxx

**Comment by danpovey** [comment]
Addressed in #1064
